### PR TITLE
TST: Fixturize tests/frame/test_missing.py

### PR DESCRIPTION
**Pull request by h-vetinari**
One more steps towards #22471.

### CODE REVIEW

File: pandas/tests/frame/test_missing.py [hunk 1]
```diff
@@ -215,42 +215,46 @@ def test_dropna_tz_aware_datetime(self):
...
         msg = "Must specify a fill 'value' or 'method'"
         with pytest.raises(ValueError, match=msg):
-            self.tsframe.fillna()
+            datetime_frame.fillna()
         msg = "Cannot specify both 'value' and 'method'"
         with pytest.raises(ValueError, match=msg):
-            self.tsframe.fillna(5, method='ffill')
+            datetime_frame.fillna(5, method='ffill')
+
+    def test_fillna_mixed_type(self, float_string_frame):
```
Comment (WillAyd) [inline pandas/tests/frame/test_missing.py]: New test here?

Comment (WillAyd) [reply pandas/tests/frame/test_missing.py]: Nevermind I see where this is coming from. I think this is OK to break out

File: pandas/tests/frame/test_missing.py [hunk 2]
```diff
@@ -215,42 +215,46 @@ def test_dropna_tz_aware_datetime(self):
...
+
+        mf = float_string_frame
+        mf.loc[mf.index[5:20], 'foo'] = np.nan
+        mf.loc[mf.index[-10:], 'A'] = np.nan
+        result = float_string_frame.fillna(value=0)
+        result = float_string_frame.fillna(method='pad')  # noqa
```
Comment (WillAyd) [inline pandas/tests/frame/test_missing.py]: Rather than noqa can just remove assignment

**Review by WillAyd — CHANGES_REQUESTED** [review]

**Comment by jreback** [comment]
lgtm ex @WillAyd comment

File: pandas/tests/frame/test_missing.py (hunk 2 above)
Comment (h-vetinari) [reply pandas/tests/frame/test_missing.py]: Well, unfortunately, this split (along fixtures) requested by @jreback is uncovering deficiencies in the existing test, in that *both* results are not tested for anything. Don't know what this should be...

**Review by h-vetinari — COMMENTED** [review]
Help appreciated.

File: pandas/tests/frame/test_missing.py (hunk 2 above)
Comment (WillAyd) [reply pandas/tests/frame/test_missing.py]: Sure entirely possible to have some mistakes in the existing tests. I'd suggest adding a TODO here to make a stronger assertion and opening a follow up issue. 

For time being OK to remove lhs and noqa note; at the very least I suppose this tests that the method doesn't raise

**Review by WillAyd — APPROVED** [review]

**Comment by WillAyd** [comment]
Thanks @h-vetinari !

**Pull request merged**